lib/supply/client.rb

```diff
@@ -18,38 +20,43 @@ class Client
     # @!group Login
     #####################################################
 
-    # Initializes the auth_client and api_client using the specified information
-    # @param path_to_key: The path to your p12 file
-    # @param issuer: Email addresss for oauth
-    # @param passphrase: Passphrase for the p12 file
-    def initialize(path_to_key: nil, issuer: nil, passphrase: nil)
-      passphrase ||= "notasecret"
+    # instanciate a client given the supplied configuration
+    def self.make_from_config
+      return Client.new(path_to_key: Supply.config[:key],
+                        issuer: Supply.config[:issuer],
+                        path_to_service_account_json: Supply.config[:json_key])
+    end
 
-      key = Google::APIClient::KeyUtils.load_from_pkcs12(File.expand_path(path_to_key), passphrase)
+    # Initializes the android_publisher and its auth_client using the specified information
+    # @param path_to_service_account_json: The path to your service account Json file
+    # @param path_to_key: The path to your p12 file (@deprecated)
+    # @param issuer: Email addresss for oauth (@deprecated)
+    def initialize(path_to_key: nil, issuer: nil, path_to_service_account_json: nil)
+      scope = Androidpublisher::AUTH_ANDROIDPUBLISHER
 
-      begin
-        self.auth_client = Signet::OAuth2::Client.new(
-          token_credential_uri: 'https://accounts.google.com/o/oauth2/token',
-          audience: 'https://accounts.google.com/o/oauth2/token',
-          scope: 'https://www.googleapis.com/auth/androidpublisher',
-          issuer: issuer,
-          signing_key: key
-        )
-      rescue => ex
-        Helper.log.fatal ex
-        raise "Authentification unsuccessful, make sure to pass a valid key file".red
+      if path_to_service_account_json
+        key_io = File.open(path_to_service_account_json)
+      else
+        require 'google/api_client/auth/key_utils'
+        key = Google::APIClient::KeyUtils.load_from_pkcs12(File.expand_path(path_to_key), 'notasecret')
```

Comment:
`passphrase` was passed to the constructor but:
1. Google documentation states the password is the same for all (https://support.google.com/cloud/answer/6158849?hl=en#serviceaccounts)
2. none of the constructors made use of it (https://github.com/fastlane/supply/blob/master/lib/supply/setup.rb#L99, https://github.com/fastlane/supply/blob/master/lib/supply/uploader.rb#L114)
3. it wasn't exposed to the CLI nor the fastlane config file

So I :skull:'ed it.